pkg/ruler/ruler.go

```diff
@@ -750,12 +854,17 @@ func (r *Ruler) GetRules(ctx context.Context, rulesRequest RulesRequest) ([]*Gro
 		return r.getShardedRules(ctx, userID, rulesRequest)
 	}
 
-	return r.getLocalRules(userID, rulesRequest)
+	return r.getLocalRules(userID, rulesRequest, false)
 }
 
-func (r *Ruler) getLocalRules(userID string, rulesRequest RulesRequest) ([]*GroupStateDesc, error) {
+func (r *Ruler) getLocalRules(userID string, rulesRequest RulesRequest, includeBackups bool) ([]*GroupStateDesc, error) {
 	groups := r.manager.GetRules(userID)
 
+	if includeBackups {
+		backupGroups := r.manager.GetBackupRules(userID)
```

Comment:
why backup is only considered in `getLocalRules`? I could not understand the code path that leads to the usage of the backup in the case sharding is used.

Reply:
`getLocalRules` when sharding is enabled will be called by by the [Rules](https://github.com/cortexproject/cortex/blob/master/pkg/ruler/ruler.go#L941-L953) method of the ruler which is the handler function by the grpc call [client.Rules](https://github.com/cortexproject/cortex/blob/master/pkg/ruler/ruler.go#L920-L925) that is called inside the [getShardedRules](https://github.com/cortexproject/cortex/blob/master/pkg/ruler/ruler.go#L887C17-L925) which is called by the [GetRules](https://github.com/cortexproject/cortex/blob/master/pkg/ruler/ruler.go#L746-L757) method that is called by the [PrometheusRules](https://github.com/cortexproject/cortex/blob/master/pkg/ruler/api.go#L161-L190) and [PrometheusAlerts](https://github.com/cortexproject/cortex/blob/master/pkg/ruler/api.go#L273-L286) API handler.